src/Commands/Site/ListCommand.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace Pantheon\Terminus\Commands\Site;
+
+use Consolidation\OutputFormatters\StructuredData\RowsOfFields;
+use Pantheon\Terminus\Commands\TerminusCommand;
+use Terminus\Collections\Sites;
+
+class ListCommand extends TerminusCommand
+{
+    /**
+     * Looks up a site name
+     *
+     * @command site:list
+     * @alias sites
+     *
+     * @field-labels
+     *   name: Name
+     *   id: ID
+     *   service_level: Service Level
+     *   framework: Framework
+     *   owner: Owner
+     *   created: Created
+     *   memberships: Memberships
+     * @option team Filter for sites you are a team member of
+     * @option owner Filter for sites a specific user owns. Use "me" for your own user
+     * @option org Filter sites you can access via the organization. Use "all" to get all.
+     * @option name Filter sites you can acces via name.
+     * @usage terminus site:list
+     *   * Responds with list of every site you can access
```

Comment:
Currently, Symfony does not support a way to attach descriptions to `usage` information.  We might want to consider what syntax / markup we want to use here, looking ahead to the time when we might have a custom help command to show this information.  Most command have no special characters here, site:lookup being the one other exception that uses `*`, like here.